aws/httpServer.ts

```diff
@@ -38,16 +39,28 @@ export class HttpServer extends pulumi.ComponentResource implements cloud.HttpSe
         // function so that we can create the underlying server once, and then call into it with
         // each request we get.
         const entryPointFactory: lambda.CallbackFactory<x.Request, x.Response> = () => {
-            // Pass */* as the binary mime types.  This tells aws-serverless-express to effectively
-            // treat all messages as binary and not reinterpret them.
-            const server = serverlessExpress.createServer(
-                createRequestListener(), /*serverListenCallback*/ undefined, /*binaryMimeTypes*/ ["*/*"]);
+            const serverPromiseFactory: () => Promise<Server> = async () => {
+                // This allows us to safely treat the value as a Promise regardless of whether
+                // or not is was passed in as an actual Promise or a synchronous factory
+                const requestListener = await Promise.resolve(createRequestListener);
+
+                // Pass */* as the binary mime types.  This tells aws-serverless-express to effectively
+                // treat all messages as binary and not reinterpret them.
+                const server = serverlessExpress.createServer(
+                    requestListener(), /*serverListenCallback*/ undefined, /*binaryMimeTypes*/ ["*/*"]);
+
+                return server;
+            };
+
+            const serverPromise = serverPromiseFactory();
```

Comment:
don't really need the helper function it doesn't seem like.  this could just be: `Promise.resolve(createRequestListener).then(rl => serverlessExpress.createServer(rl()...)` right?